### DISCUSSION TITLE
Do you plan to update the version to react-floater v0.8.x?

### DISCUSSION THREAD

**Discussion by Himenon** [gilbarbara/react-joyride#820] [Q&A] [answered]
Thanks for the Great Library.

I have confirmed that bundling by vite is now possible by using v0.8.x of react-floater.

* https://github.com/gilbarbara/react-floater/issues/83#issuecomment-997308489

I know that this version is not yet incorporated and available in react-joyride.

* https://github.com/gilbarbara/react-joyride/blob/main/package.json#L40

I was going to suggest a pull request, but I have found (My [PR](https://github.com/Himenon/react-joyride/pull/1)) several issues that cannot be easily fixed, such as test fixes due to popperjs version updates and missing [HTMLElement when initializing Scope](https://github.com/gilbarbara/react-joyride/blob/main/src/modules/scope.js#L9).

As the title says, do you plan to update the version of react-floater?

thanks.

**Comment by gilbarbara** [comment, marked as answer]
Yes, but isn't a simple task.
react-floater needs to support pre-rendering (which I'm already working on) and I also need to rewrite the tests with @testing-library since Enzyme is dead.

**Reply by Himenon** [reply]
Thanks for the response. I understand the situation.
I also found a workaround to use v0.7.x with vite and will share it with you.

```html
<head>
  <script>
    global = typeof global !== "undefined" ? global : window;
  </script>
</head>
```

I have confirmed that react-joyride@v0.7.x works by using this.

**Reply by johnhunter** [reply]
I'm also interested in an upgrade to react-floater.

I'm seeing an occasional exception `Cannot read properties of null (reading 'isConnected')`
```
if (!this.floaterRef.isConnected) {
```
https://github.com/gilbarbara/react-floater/blob/v0.7.5/src/components/index.js#L274

I've been unable to reproduce but it is reported in my production environment.

**Reply by mgolpashin** [reply]
We need to control the positioning of the Tooltip, as it needs to be a child of a specific ReactContext. Correct me if I'm wrong, but I don't think that this is possible with v0.7 of react-floater. The `portalElement` got introduced with v0.8 and [in v0.7, the portal is always appended to the document body](https://github.com/gilbarbara/react-floater/blob/v0.7/src/components/Portal.js#L65). 

Or is there any other way to archieve this?

**Reply by belinde** [reply]
Is there any change in sight? react-floater 0.7 depends on a deprecated version of popper.js, now. Nothing to be concerned about, propbably, it's just annoying to see the deprecation notice in the logs

**Reply by gilbarbara** [reply]
Hey @belinde 
I just finished refactoring the code to TypeScript, so upgrading will be much easier.
I also fixed some bugs in the react-floater, but it requires react 16.8+ which will be a breaking change for this package. 

I'm working on the new state to allow asynchronous steps and other feature requests.
I hope to get an alpha version by the end of November.

**Comment by gilbarbara** [comment]
I migrate to @floating-ui in v3.0.0

Full migration guide: https://react-joyride.com/docs/migration
What's new: https://react-joyride.com/docs/new-in-v3

Thanks
